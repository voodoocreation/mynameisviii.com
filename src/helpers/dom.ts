import getScrollbarWidth from "scrollbar-size";

if (typeof window !== "undefined") {
  window.scrollState = {
    count: 0,
    left: 0,
    top: 0
  };
}

export const isServer = () => typeof window === "undefined" || window.isServer;

export const isInViewport = (element?: HTMLElement | null) => {
  if (!element) {
    return false;
  }

  const bounds = element.getBoundingClientRect();
  const viewport = {
    height: window.innerHeight || document.documentElement.clientHeight,
    width: window.innerWidth || document.documentElement.clientWidth
  };

  return (
    bounds.top >= 0 &&
    bounds.right <= viewport.width &&
    bounds.bottom <= viewport.height &&
    bounds.left >= 0
  );
};

export const isAlmostInViewport = (
  element?: HTMLElement | null,
  offset = 0
) => {
  if (!element) {
    return false;
  }

  const bounds = element.getBoundingClientRect();
  const viewport = {
    height: window.innerHeight || document.documentElement.clientHeight,
    width: window.innerWidth || document.documentElement.clientWidth
  };

  return (
    viewport.height - bounds.top > -offset &&
    bounds.right > -offset &&
    bounds.bottom > -offset &&
    viewport.width - bounds.left > -offset
  );
};

export const lockScroll = () => {
  const { body } = document;
  const html = document.documentElement;

  window.scrollState.count += 1;

  if (!html.classList.contains("isLocked")) {
    const existingPadding = parseInt(
      getComputedStyle(body, null).getPropertyValue("padding-right"),
      10
    );
    const scrollbarWidth = getScrollbarWidth();
    const newPadding =
      isNaN(existingPadding) || !isFinite(existingPadding)
        ? scrollbarWidth
        : scrollbarWidth + existingPadding;
    const scrollElement =
      body.scrollTop > 0 || html.scrollTop === 0 ? body : html;

    window.scrollState.top = scrollElement.scrollTop;
    window.scrollState.left = scrollElement.scrollLeft;

    html.classList.add("isLocked");
    body.style.paddingRight = `${newPadding}px`;
  }
};

export const unlockScroll = () => {
  const { body } = document;
  const html = document.documentElement;

  if (window.scrollState.count <= 1 && html.classList.contains("isLocked")) {
    body.style.paddingRight = "";
    html.classList.remove("isLocked");
    window.scrollTo(window.scrollState.left, window.scrollState.top);
  }

  if (window.scrollState.count > 0) {
    window.scrollState.count -= 1;
  }
};
