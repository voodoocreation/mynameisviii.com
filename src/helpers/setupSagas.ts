import merge from "lodash.merge";
import SagaTester from "redux-saga-tester";

import rootReducer from "../reducers/root.reducers";
import rootSaga from "../sagas/root.sagas";

export default (fromTestStore: {}, fromTestPorts: {}) => {
  const initialState = merge({}, fromTestStore);
  const ports = merge(
    {
      api: {},
      dataLayer: [],
      maps: {}
    },
    fromTestPorts
  );

  const sagaTester = new SagaTester({
    initialState,
    reducers: rootReducer
  });
  sagaTester.start(rootSaga(ports));

  return {
    dispatch: (action: any) => sagaTester.dispatch(action),
    findAction: (actionFromTest: any) =>
      sagaTester
        .getCalledActions()
        .find((action: any) => action.type === actionFromTest().type),
    sagaTester,
    store: () => sagaTester.getState()
  };
};
