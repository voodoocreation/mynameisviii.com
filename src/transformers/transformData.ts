export const arrayToAssoc = (
  array: Array<{}>,
  key: string,
  failSilently = true
) =>
  array.reduce(
    (
      acc: { [index: string]: {} },
      curr: { [index: string]: any },
      index: number
    ) => {
      if (curr[key] !== undefined) {
        acc[curr[key]] = curr;
      } else if (!failSilently) {
        throw new Error(`Key '${key}' not found at index ${index}`);
      }
      return acc;
    },
    {}
  );

export const assocToArray: (object: any) => any = object =>
  Object.keys(object).reduce((acc: Array<{}>, curr: string) => {
    acc.push(object[curr]);
    return acc;
  }, []);

export const tryParseJson = (json: string | Error) => {
  let result;

  try {
    result = json.toString ? JSON.parse(json.toString()) : json;
  } catch (error) {
    result = json;
  }

  return result;
};

export const lengthToDuration = (length: string) => {
  const segments = length.split(":");

  switch (segments.length) {
    default:
    case 2:
      return `PT${segments[0]}M${segments[1]}S`;

    case 3:
      return `PT${segments[0]}H${segments[1]}M${segments[2]}S`;
  }
};

export const absUrl = (path: string) =>
  process.env.NODE_ENV !== "production"
    ? `http://localhost:${process.env.PORT || window.location.port}${path}`
    : `http://${
        typeof window === "undefined"
          ? process.env.DOMAIN
          : window.location.host
      }${path}`;

export const extractDomain = (url: string) => url.split("/")[2];
