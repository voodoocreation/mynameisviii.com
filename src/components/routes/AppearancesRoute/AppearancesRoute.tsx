import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { FormattedMessage, InjectedIntl } from "react-intl";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import { ActionCreator } from "typescript-fsa";

import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import { absUrl } from "../../../transformers/transformData";
import AppearanceListing from "../../presentation/AppearanceListing/AppearanceListing";
import ButtonBar from "../../presentation/ButtonBar/ButtonBar";
import LoadButton from "../../presentation/LoadButton/LoadButton";
import NoResults from "../../presentation/NoResults/NoResults";
import PageHeader from "../../presentation/PageHeader/PageHeader";

import * as actions from "../../../actions/root.actions";
import * as selectors from "../../../selectors/root.selectors";

interface IStoreProps {
  appearancesCount: number;
  hasAllAppearances: boolean;
  isLoading: boolean;
  pastAppearances: IAppearance[];
  upcomingAppearances: IAppearance[];
}

interface IDispatchProps {
  fetchAppearances: ActionCreator<{}>;
  fetchMoreAppearances: ActionCreator<{}>;
}

interface IProps extends IStoreProps, IDispatchProps {
  intl: InjectedIntl;
}

interface IState {
  loadedListings: {
    [index: string]: boolean;
  };
}

class AppearancesRoute extends React.Component<IProps, IState> {
  public static async getInitialProps(props: any) {
    const { isServer, store } = props.ctx;

    const state = store.getState();

    if (isServer || selectors.getAppearancesCount(state) === 0) {
      store.dispatch(actions.fetchAppearances.started({}));
    } else if (!selectors.getHasAllAppearances(state)) {
      store.dispatch(actions.fetchMoreAppearances.started({}));
    }
  }

  public readonly state = {
    loadedListings: {}
  };

  public render() {
    const {
      appearancesCount,
      hasAllAppearances,
      pastAppearances,
      upcomingAppearances
    } = this.props;
    const { formatMessage } = this.props.intl;

    const hasLoadedAllListings =
      appearancesCount === Object.keys(this.state.loadedListings).length;

    const isLoading = this.props.isLoading || !hasLoadedAllListings;

    const loadMoreButton = hasAllAppearances ? null : (
      <LoadButton isLoading={isLoading} onLoad={this.onLoadMore}>
        <FormattedMessage id="LOAD_MORE" />
      </LoadButton>
    );

    return (
      <article className={cn("AppearancesRoute", { hasLoadedAllListings })}>
        <Head>
          <title>
            {formatMessage({ id: "APPEARANCES_TITLE" })}
            {" Â· "}
            {formatMessage({ id: "BRAND_NAME" })}
          </title>

          <meta
            content={formatMessage({ id: "APPEARANCES_DESCRIPTION" })}
            name="description"
          />

          <meta
            property="og:title"
            content={formatMessage({ id: "APPEARANCES_TITLE" })}
          />
          <meta
            property="og:description"
            content={formatMessage({ id: "APPEARANCES_DESCRIPTION" })}
          />
          <meta
            property="og:image"
            content="https://s3.amazonaws.com/mynameisviii-static/homepage-og.jpg"
          />
          <meta property="og:url" content={absUrl("/appearances")} />
          <meta property="og:type" content="website" />
        </Head>

        <PageHeader>
          <FormattedMessage id="APPEARANCES_TITLE" />
        </PageHeader>

        {upcomingAppearances.length > 0 ? (
          <section className="AppearanceListings AppearanceListings-upcoming">
            <h2>
              <FormattedMessage id="UPCOMING" />
            </h2>

            <div className="AppearanceListings-items">
              {upcomingAppearances.map(appearance => (
                <AppearanceListing
                  key={appearance.slug}
                  onLoad={this.onListingLoad}
                  {...appearance}
                />
              ))}
            </div>
          </section>
        ) : null}

        {pastAppearances.length > 0 ? (
          <section className="AppearanceListings AppearanceListings-past">
            <h2>
              <FormattedMessage id="PAST" />
            </h2>

            <div className="AppearanceListings-items">
              {pastAppearances.map(appearance => (
                <AppearanceListing key={appearance.slug} {...appearance} />
              ))}
            </div>
          </section>
        ) : null}

        {hasAllAppearances &&
        upcomingAppearances.length + pastAppearances.length === 0 ? (
          <NoResults>
            <p>
              <FormattedMessage id="NO_APPEARANCES" />
            </p>
          </NoResults>
        ) : null}

        <ButtonBar>{loadMoreButton}</ButtonBar>
      </article>
    );
  }

  private onListingLoad = (slug: string) => {
    this.setState({
      loadedListings: {
        ...this.state.loadedListings,
        [slug]: true
      }
    });
  };

  private onLoadMore = () => {
    this.props.fetchMoreAppearances({});
  };
}

const mapStateToProps = (state: any) => ({
  appearancesCount: selectors.getAppearancesCount(state),
  hasAllAppearances: selectors.getHasAllAppearances(state),
  isLoading: selectors.getAppearancesIsLoading(state),
  pastAppearances: selectors.getPastAppearances(state),
  upcomingAppearances: selectors.getUpcomingAppearances(state)
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchAppearances: actions.fetchAppearances.started,
      fetchMoreAppearances: actions.fetchMoreAppearances.started
    },
    dispatch
  );

export default injectIntlIntoPage(
  connect<IStoreProps, IDispatchProps>(
    mapStateToProps,
    mapDispatchToProps
  )(AppearancesRoute)
);
