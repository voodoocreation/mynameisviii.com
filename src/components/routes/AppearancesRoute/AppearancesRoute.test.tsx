import { mount, render } from "enzyme";
import { camelizeKeys } from "humps";
import * as React from "react";
import { Provider } from "react-redux";

import * as selectors from "../../../selectors/root.selectors";
import createStore from "../../../store/root.store";
import AppearancesRoute from "./AppearancesRoute";

import appearances from "../../../../server/mocks/appearances.json";
import * as dom from "../../../helpers/dom";
import { arrayToAssoc } from "../../../transformers/transformData";

const mockData: any = camelizeKeys(appearances);

const setup = (fn: any, fromTestStore = {}, fromTestApi?: {}) => {
  const store = createStore(fromTestStore, {}, fromTestApi);

  return {
    actual: fn(
      <Provider store={store}>
        <AppearancesRoute />
      </Provider>
    ),
    store
  };
};

describe("[routes] <AppearancesRoute />", () => {
  beforeAll(() => {
    Object.defineProperty(dom, "isAlmostInViewport", {
      value: jest.fn(() => false)
    });
  });

  it("renders correctly", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: true,
        items: arrayToAssoc(mockData.items, "slug")
      }
    });

    expect(actual).toMatchSnapshot();
  });

  it("renders upcoming appearances correctly", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: true,
        items: {
          "test-1": {
            ...mockData.items[0],
            finishingAt: "2020-01-01T00:00:00"
          }
        }
      }
    });

    expect(actual.find(".AppearanceListings-upcoming")).toHaveLength(1);
  });

  it("renders past appearances correctly", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: true,
        items: {
          "test-1": {
            ...mockData.items[0],
            finishingAt: "2000-01-01T00:00:00"
          }
        }
      }
    });

    expect(actual.find(".AppearanceListings-past")).toHaveLength(1);
  });

  it("renders no results correctly", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: true
      }
    });

    expect(actual.find(".NoResults")).toHaveLength(1);
  });

  it("renders with an error correctly", () => {
    const { actual } = setup(render, {
      appearances: {
        error: {
          message: "Server error",
          status: 500
        }
      }
    });

    expect(actual.find(".LoadButton").text()).toBe("Try again");
    expect(actual).toMatchSnapshot();
  });

  it("renders with 'load more' button when hasAllItems=false", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: false
      }
    });

    expect(actual.find(".LoadButton")).toHaveLength(1);
  });

  it("renders without 'load more' button when hasAllItems=true", () => {
    const { actual } = setup(render, {
      appearances: {
        hasAllItems: true
      }
    });

    expect(actual.find(".LoadButton")).toHaveLength(0);
  });

  it("loads more listings when 'load more' button is activated", () => {
    const { actual } = setup(
      mount,
      {},
      {
        fetchAppearances: () => ({
          data: {
            items: [mockData.items[0]]
          },
          ok: true
        })
      }
    );

    expect(actual.find("AppearanceListing")).toHaveLength(0);
    actual.find("LoadButton").simulate("click");
    expect(actual.find("AppearanceListing")).toHaveLength(1);
  });

  it("updates `loadedListings` state after a listing has loaded", () => {
    const { actual } = setup(mount, {
      appearances: {
        items: { "test-1": mockData.items[0] }
      }
    });

    actual.find("AppearanceListing img").simulate("load");

    expect(actual.find(".hasLoadedAllListings")).toHaveLength(1);
  });

  describe("getInitialProps()", () => {
    const items = [mockData.items[0]];

    it("fetches items when rendering on the server", async () => {
      const { store } = setup(
        render,
        {},
        {
          fetchAppearances: () => ({
            data: { items },
            ok: true
          })
        }
      );

      await AppearancesRoute.getInitialProps({
        ctx: {
          isServer: true,
          store
        }
      });

      expect(selectors.getAppearancesAsArray(store.getState())).toEqual(items);
    });

    it("fetches items when the store is empty", async () => {
      const { store } = setup(
        render,
        {},
        {
          fetchAppearances: () => ({
            data: { items },
            ok: true
          })
        }
      );

      await AppearancesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getAppearancesAsArray(store.getState())).toEqual(items);
    });

    it("fetches more items when there are more to fetch", async () => {
      const { store } = setup(
        render,
        {
          appearances: {
            items: {
              "test-1": items[0]
            }
          }
        },
        {
          fetchAppearances: () => ({
            data: {
              items: [{ slug: "test-2" }]
            },
            ok: true
          })
        }
      );

      expect(selectors.getAppearancesCount(store.getState())).toBe(1);

      await AppearancesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getAppearancesCount(store.getState())).toBe(2);
    });

    it("doesn't fetch anything when all items are in the store", async () => {
      const { store } = setup(render, {
        appearances: {
          hasAllItems: true,
          items: {
            "test-1": items[0]
          }
        }
      });

      await AppearancesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getAppearancesAsArray(store.getState())).toEqual(items);
    });
  });
});
