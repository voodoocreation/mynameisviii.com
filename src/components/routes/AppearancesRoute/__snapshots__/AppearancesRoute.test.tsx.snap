// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <AppearancesRoute /> renders correctly 1`] = `
<article
  class="AppearancesRoute"
>
  <header
    class="PageHeader"
  >
    <h1>
      <span>
        Appearances
      </span>
    </h1>
  </header>
  <section
    class="AppearanceListings AppearanceListings-upcoming"
  >
    <h2>
      <span>
        Upcoming
      </span>
    </h2>
    <div
      class="AppearanceListings-items"
    >
      <article
        class="AppearanceListing"
      >
        <a
          href="/appearances/nothing-but-yet-another-test-gig-2018-11-13"
        >
          <div
            class="AppearanceListing-details"
          >
            <header
              class="AppearanceListing-header"
            >
              <h3>
                Nothing But Yet Another Test Gig
              </h3>
            </header>
            <section
              class="AppearanceListing-meta"
            >
              <div
                class="Meta AppearanceListing-date"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m31.6 33.4v-18.4h-23.2v18.4h23.2z m0-26.8c1.8 0 3.4 1.6 3.4 3.4v23.4c0 1.8-1.6 3.2-3.4 3.2h-23.2c-1.9 0-3.4-1.4-3.4-3.2v-23.4c0-1.8 1.5-3.4 3.4-3.4h1.6v-3.2h3.4v3.2h13.2v-3.2h3.4v3.2h1.6z m-3.2 11.8v3.2h-3.4v-3.2h3.4z m-6.8 0v3.2h-3.2v-3.2h3.2z m-6.6 0v3.2h-3.4v-3.2h3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Date
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-01-20T07:00:00.000Z"
                    title="January 20, 2018"
                  >
                    January 20, 2018
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-time"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8q5.5 0 9.4-4t4-9.4-4-9.4-9.4-4-9.4 4-4 9.4 4 9.4 9.4 4z m0-30q6.9 0 11.8 4.8t4.8 11.8-4.8 11.8-11.8 4.8-11.8-4.8-4.8-11.8 4.8-11.8 11.8-4.8z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Time
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-01-20T07:00:00.000Z"
                    title="8:00 PM"
                  >
                    8:00 PM
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-location"
                title="Meow - 9 Edward Street, Te Aro, Wellington 6011"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20 19.1q1.7 0 2.9-1.2t1.2-2.9-1.2-2.9-2.9-1.2-2.9 1.2-1.2 2.9 1.2 2.9 2.9 1.2z m0-15.7q4.8 0 8.2 3.4t3.4 8.2q0 2.4-1.2 5.5t-2.9 5.9-3.4 5.1-2.8 3.8l-1.3 1.3q-0.5-0.5-1.2-1.4t-2.9-3.6-3.5-5.2-2.8-5.8-1.2-5.6q0-4.8 3.4-8.2t8.2-3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Location
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  Meow
                </span>
              </div>
              <div
                class="Meta AppearanceListing-audience"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m26.6 21.6c3.9 0 11.8 2 11.8 5.9v4.1h-10v-4.1c0-2.5-1.4-4.4-3.3-5.8 0.5-0.1 1.1-0.1 1.5-0.1z m-13.2 0c3.9 0 11.6 2 11.6 5.9v4.1h-23.4v-4.1c0-3.9 7.9-5.9 11.8-5.9z m0-3.2c-2.8 0-5-2.3-5-5s2.2-5 5-5 4.9 2.2 4.9 5-2.2 5-4.9 5z m13.2 0c-2.7 0-5-2.3-5-5s2.3-5 5-5 5 2.2 5 5-2.2 5-5 5z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Audience
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  All ages
                </span>
              </div>
              <div
                class="Meta AppearanceListing-price"
              >
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Price
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <span
                    class="Price PriceRange-min"
                  >
                    <span>
                      NZ$5
                    </span>
                  </span>
                  â€“
                  <span
                    class="Price PriceRange-max"
                  >
                    <span>
                      NZ$20
                    </span>
                  </span>
                </span>
              </div>
            </section>
          </div>
          <figure
            class="AppearanceListing-image"
          >
            <img
              alt="Nothing But Yet Another Test Gig"
              src="https://s3.amazonaws.com/mynameisviii-static/appearances/nothing-but-a-test-gig-2018-10-20.jpg"
            />
          </figure>
        </a>
      </article>
      <article
        class="AppearanceListing"
      >
        <a
          href="/appearances/nothing-but-a-test-gig-2018-11-13"
        >
          <div
            class="AppearanceListing-details"
          >
            <header
              class="AppearanceListing-header"
            >
              <h3>
                Nothing But a Test Gig
              </h3>
            </header>
            <section
              class="AppearanceListing-meta"
            >
              <div
                class="Meta AppearanceListing-date"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m31.6 33.4v-18.4h-23.2v18.4h23.2z m0-26.8c1.8 0 3.4 1.6 3.4 3.4v23.4c0 1.8-1.6 3.2-3.4 3.2h-23.2c-1.9 0-3.4-1.4-3.4-3.2v-23.4c0-1.8 1.5-3.4 3.4-3.4h1.6v-3.2h3.4v3.2h13.2v-3.2h3.4v3.2h1.6z m-3.2 11.8v3.2h-3.4v-3.2h3.4z m-6.8 0v3.2h-3.2v-3.2h3.2z m-6.6 0v3.2h-3.4v-3.2h3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Date
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-10-20T07:00:00.000Z"
                    title="October 20, 2018"
                  >
                    October 20, 2018
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-time"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8q5.5 0 9.4-4t4-9.4-4-9.4-9.4-4-9.4 4-4 9.4 4 9.4 9.4 4z m0-30q6.9 0 11.8 4.8t4.8 11.8-4.8 11.8-11.8 4.8-11.8-4.8-4.8-11.8 4.8-11.8 11.8-4.8z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Time
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-10-20T07:00:00.000Z"
                    title="8:00 PM"
                  >
                    8:00 PM
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-location"
                title="Meow - 9 Edward Street, Te Aro, Wellington 6011"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20 19.1q1.7 0 2.9-1.2t1.2-2.9-1.2-2.9-2.9-1.2-2.9 1.2-1.2 2.9 1.2 2.9 2.9 1.2z m0-15.7q4.8 0 8.2 3.4t3.4 8.2q0 2.4-1.2 5.5t-2.9 5.9-3.4 5.1-2.8 3.8l-1.3 1.3q-0.5-0.5-1.2-1.4t-2.9-3.6-3.5-5.2-2.8-5.8-1.2-5.6q0-4.8 3.4-8.2t8.2-3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Location
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  Meow
                </span>
              </div>
              <div
                class="Meta AppearanceListing-audience"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m26.6 21.6c3.9 0 11.8 2 11.8 5.9v4.1h-10v-4.1c0-2.5-1.4-4.4-3.3-5.8 0.5-0.1 1.1-0.1 1.5-0.1z m-13.2 0c3.9 0 11.6 2 11.6 5.9v4.1h-23.4v-4.1c0-3.9 7.9-5.9 11.8-5.9z m0-3.2c-2.8 0-5-2.3-5-5s2.2-5 5-5 4.9 2.2 4.9 5-2.2 5-4.9 5z m13.2 0c-2.7 0-5-2.3-5-5s2.3-5 5-5 5 2.2 5 5-2.2 5-5 5z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Audience
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  18+
                </span>
              </div>
              <div
                class="Meta AppearanceListing-price"
              >
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Price
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <span
                    class="Price PriceRange-min"
                  >
                    <span>
                      NZ$10
                    </span>
                  </span>
                  â€“
                  <span
                    class="Price PriceRange-max"
                  >
                    <span>
                      NZ$20
                    </span>
                  </span>
                </span>
              </div>
            </section>
          </div>
          <figure
            class="AppearanceListing-image"
          >
            <img
              alt="Nothing But a Test Gig"
              src="https://s3.amazonaws.com/mynameisviii-static/appearances/nothing-but-a-test-gig-2018-10-20.jpg"
            />
          </figure>
        </a>
      </article>
      <article
        class="AppearanceListing isCancelled"
      >
        <a
          href="/appearances/nothing-but-another-test-gig-2018-11-13"
        >
          <div
            class="AppearanceListing-details"
          >
            <header
              class="AppearanceListing-header"
            >
              <h3>
                Nothing But Another Test Gig
              </h3>
            </header>
            <section
              class="AppearanceListing-meta"
            >
              <div
                class="Meta AppearanceListing-status"
              >
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Status
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <span>
                    Cancelled
                  </span>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-date"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m31.6 33.4v-18.4h-23.2v18.4h23.2z m0-26.8c1.8 0 3.4 1.6 3.4 3.4v23.4c0 1.8-1.6 3.2-3.4 3.2h-23.2c-1.9 0-3.4-1.4-3.4-3.2v-23.4c0-1.8 1.5-3.4 3.4-3.4h1.6v-3.2h3.4v3.2h13.2v-3.2h3.4v3.2h1.6z m-3.2 11.8v3.2h-3.4v-3.2h3.4z m-6.8 0v3.2h-3.2v-3.2h3.2z m-6.6 0v3.2h-3.4v-3.2h3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Date
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-10-20T07:00:00.000Z"
                    title="October 20, 2018"
                  >
                    October 20, 2018
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-time"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8q5.5 0 9.4-4t4-9.4-4-9.4-9.4-4-9.4 4-4 9.4 4 9.4 9.4 4z m0-30q6.9 0 11.8 4.8t4.8 11.8-4.8 11.8-11.8 4.8-11.8-4.8-4.8-11.8 4.8-11.8 11.8-4.8z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Time
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <time
                    class="DateTime"
                    datetime="2018-10-20T07:00:00.000Z"
                    title="8:00 PM"
                  >
                    8:00 PM
                  </time>
                </span>
              </div>
              <div
                class="Meta AppearanceListing-location"
                title="Meow - 9 Edward Street, Te Aro, Wellington 6011"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20 19.1q1.7 0 2.9-1.2t1.2-2.9-1.2-2.9-2.9-1.2-2.9 1.2-1.2 2.9 1.2 2.9 2.9 1.2z m0-15.7q4.8 0 8.2 3.4t3.4 8.2q0 2.4-1.2 5.5t-2.9 5.9-3.4 5.1-2.8 3.8l-1.3 1.3q-0.5-0.5-1.2-1.4t-2.9-3.6-3.5-5.2-2.8-5.8-1.2-5.6q0-4.8 3.4-8.2t8.2-3.4z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Location
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  Meow
                </span>
              </div>
              <div
                class="Meta AppearanceListing-audience"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m26.6 21.6c3.9 0 11.8 2 11.8 5.9v4.1h-10v-4.1c0-2.5-1.4-4.4-3.3-5.8 0.5-0.1 1.1-0.1 1.5-0.1z m-13.2 0c3.9 0 11.6 2 11.6 5.9v4.1h-23.4v-4.1c0-3.9 7.9-5.9 11.8-5.9z m0-3.2c-2.8 0-5-2.3-5-5s2.2-5 5-5 4.9 2.2 4.9 5-2.2 5-4.9 5z m13.2 0c-2.7 0-5-2.3-5-5s2.3-5 5-5 5 2.2 5 5-2.2 5-5 5z"
                    />
                  </g>
                </svg>
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Audience
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  18+
                </span>
              </div>
              <div
                class="Meta AppearanceListing-price"
              >
                 
                <span
                  class="Meta-label"
                >
                  <span>
                    Price
                  </span>
                  :
                </span>
                 
                <span
                  class="Meta-value"
                >
                  <span
                    class="Price PriceRange-min"
                  >
                    <span>
                      NZ$20
                    </span>
                  </span>
                </span>
              </div>
            </section>
          </div>
          <figure
            class="AppearanceListing-image"
          >
            <img
              alt="Nothing But Another Test Gig"
              src="https://s3.amazonaws.com/mynameisviii-static/appearances/nothing-but-a-test-gig-2018-10-20.jpg"
            />
          </figure>
        </a>
      </article>
    </div>
  </section>
</article>
`;
